runtime/jcl/common/criusupport.c

```diff
@@ -41,6 +103,200 @@ Java_com_ibm_jvm_CRIUSupport_isCRIUSupportEnabledImpl(JNIEnv *env, jclass unused
+	}
+
+	/* get the required size */
+	requiredConvertedStringSize = omrstr_convert(J9STR_CODE_MUTF8, J9STR_CODE_PLATFORM_RAW,
+					mutf8String,
+					mutf8StringBufSize,
+					*nativeString,
+					0);
+
+	if (requiredConvertedStringSize < 0) {
+		vmFuncs->setCurrentExceptionNLSWithArgs(currentThread, J9NLS_JCL_CRIU_FAILED_TO_CONVERT_JAVA_STRING, J9VMCONSTANTPOOL_JAVALANGINTERNALERROR, requiredConvertedStringSize);
+		*resultType = vm->criuSupportInvalidArguments;
+		res = FALSE;
+		goto free;
+	}
+
+	if (requiredConvertedStringSize > STRING_BUFFER_SIZE) {
+		*nativeString = (char*)j9mem_allocate_memory(requiredConvertedStringSize, OMRMEM_CATEGORY_VM);
+		if (NULL == *nativeString) {
+			vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
+			*resultType = vm->criuSupportJVMCheckpointFailure;
+			res = FALSE;
+			goto free;
+		}
+	}
+
+	/* convert the string */
+	requiredConvertedStringSize = omrstr_convert(J9STR_CODE_MUTF8, J9STR_CODE_PLATFORM_RAW,
+					mutf8String,
+					mutf8StringBufSize,
+					*nativeString,
+					requiredConvertedStringSize);
+
+	if (requiredConvertedStringSize < 0) {
+		vmFuncs->setCurrentExceptionNLSWithArgs(currentThread, J9NLS_JCL_CRIU_FAILED_TO_CONVERT_JAVA_STRING, J9VMCONSTANTPOOL_JAVALANGINTERNALERROR, requiredConvertedStringSize);
+		*resultType = vm->criuSupportInvalidArguments;
+		res = FALSE;
+		goto free;
+	}
+
+free:
+	if (mutf8String != mutf8StringBuf) {
+		j9mem_free_memory(mutf8String);
+	}
+
+	return res;
+}
+
+
+jobject JNICALL
+Java_org_eclipse_openj9_CRIUSupport_checkpointJVMImpl(JNIEnv *env,
+							jclass unused,
+							jstring checkPointDir,
+							jboolean keepRunning,
+							jboolean shellJob,
+							jboolean extUnixSupport,
+							jint logLevel,
+							jstring logFile)
+{
+	J9VMThread *currentThread = (J9VMThread*)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+
+	jthrowable currentExceptionLocalRef = NULL;
+	jfieldID resultType = vm->criuSupportUnsupportedOperation;
+
+	if (J9_ARE_ALL_BITS_SET(vm->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_ENABLE_CRIU_SUPPORT)) {
+#if defined(LINUX)
+		j9object_t cpDir = NULL;
+		j9object_t log = NULL;
+		IDATA dirFD = 0;
+		char directoryBuf[STRING_BUFFER_SIZE];
+		char *directoryChars = (char*)directoryBuf;
+		char logFileBuf[STRING_BUFFER_SIZE];
+		char *logFileChars = logFileBuf;
+		IDATA systemReturnCode = 0;
+		PORT_ACCESS_FROM_VMC(currentThread);
+
+		memset(directoryBuf, 0, sizeof(directoryBuf));
+		memset(logFileBuf, 0, sizeof(logFileBuf));
+
+		vmFuncs->internalEnterVMFromJNI(currentThread);
+
+		cpDir = J9_JNI_UNWRAP_REFERENCE(checkPointDir);
+		log = J9_JNI_UNWRAP_REFERENCE(logFile);
+
+		if (FALSE == getNativeString(currentThread, cpDir, &directoryChars, (char**)&directoryBuf, &resultType)) {
+			/* errors and return codes are set in the helper */
+			goto freeDir;
+		}
+
+		if (FALSE == getNativeString(currentThread, log, &logFileChars, (char**)&logFileBuf, &resultType)) {
```

Comment:
These calls are just wrong; the clue is the need for the casts. The penultimate arguments should be `&directoryChars` and `&logFileChars`, respectively.